src/platforms/dotnet/guides/aspnetcore/performance/included-instrumentation.mdx

```diff
@@ -0,0 +1,137 @@
+        // Make sure to put this middleware right after `UseRouting()`.
+        app.UseSentryTracing();
+
+        app.UseAuthorization();
+
+        app.UseEndpoints(endpoints =>
+        {
+            endpoints.MapControllerRoute(
+                name: "default",
+                pattern: "{controller=Home}/{action=Index}/{id?}");
+        });
+    }
+}
+```
+
+Transaction names follow the pattern `<HTTP method> <Route>`; for example, a request to the following action will create a transaction named `GET /person/{id}`:
+
+```csharp
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+public class HomeController : Controller
+{
+    [HttpGet("/person/{id}")]
+    public IActionResult Person(string id) { /* ... */ }
+}
+```
+
+Transactions created by this integration are automatically added to the current scope. If you want to start your own spans from the automatically created transaction, you can use the following approach:
+
+```csharp
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Sentry;
+
+public class HomeController : Controller
+{
+    private readonly IHub _sentryHub;
+
+    public HomeController(IHub sentryHub) => _sentryHub = sentryHub;
+
+    [HttpGet("/person/{id}")]
+    public IActionResult Person(string id)
+    {
+        // Get the last non-finished span (or transaction).
+        // This can generally be null, but since the root transaction
+        // is set automatically by the middleware, we can expect it
+        // to always be defined.
+        var currentSpan = hub.GetSpan();
+
+        // We can then start our own spans which will be attached to
+        // the automatically created transaction.
+        var childSpan = currentSpan?.StartChild("additional-work");
+
+        // ...
+
+        childSpan?.Finish();
+
+        // Note: don't call `currentSpan.Finish()` here as it may prematurely
+        // finish the root transaction and send it to Sentry before the
+        // request has fully completed.
+    }
+}
+```
+
+### HTTP Client Integration
+
+Sentry also provides a custom filter for HTTP client factory commonly used in ASP.NET Core applications. This filter adds an additional message handler that automatically injects Sentry's trace header and tracks outgoing HTTP requests in separate spans attached to the current transaction.
+
+To enable this integration, you need to have HTTP client factory enabled within your application:
+
+```csharp
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+
+public class Startup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        /* ... */
+
+        services.AddRouting();
+
+        // This is required for HTTP client integration
+        services.AddHttpClient();
+    }
+}
+```
+
+If you already have `AddHttpClient()` in your service collection (which is likely the case), then no action is required.
```

Comment:
Should we document how to use the `HttpMessageHandler`? 
I guess that could be part of the core .NET SDK anyway

Reply:
Yeah. I'm not sure where to put it exactly. Do you have a suggestion?

Reply:
Another `Included Instrumentation` section like this one, but on the main dotnet level.

We could name it: `Propagating a trace`.

And add a note like:

> If you are using ASP.NET Core integration this is done automatically. Check the Included Insrumentation docs for ASP.NET Core

Reply:
Can I create an issue in this repo so I don't forget?

Reply:
I'll add u to this repo wait up